File: Fix.Dictionary/Pedigree.cs
using System;
using System.Collections.Generic;

namespace Fix;

public static partial class Dictionary
{

    public record Pedigree
    {
        public string? Added { get; init; }
        public string? AddedEP { get; init; }
        public string? Updated { get; init; }
        public string? UpdatedEP { get; init; }
        public string? Deprecated { get; init; }
        public string? DeprecatedEP { get; init; }

        public static Pedigree Empty { get; } = new Pedigree();

        public override string ToString()
        {
            var items = new List<String>();

            if (Added is string added)
            {
                items.Add($"Added = {added}");
            }

            if (AddedEP is string addedEP)
            {
                items.Add($"AddedEP = {addedEP}");
            }

            if (Updated is string updated)
            {
                items.Add($"Updated = {updated}");
            }

            if (UpdatedEP is string updatedEP)
            {
                items.Add($"UpdatedEP = {updatedEP}");
            }

            if (Deprecated is string deprecated)
            {
                items.Add($"Deprecated = {deprecated}");
            }

            if (DeprecatedEP is string deprecatedEP)
            {
                items.Add($"DeprecatedEP = {deprecatedEP}");
            }

            if (items.Count == 0)
            {
                return string.Empty;
            }

            return string.Join(", ", items);
        }
    }
}

File: Fix.Lexicographer/pedigree.py
#!/usr/bin/python3

def format_pedigree(pedigree):
    initialisers = []
    if pedigree.added:
        initialisers.append('Added = "{}"'.format(pedigree.added))
    if pedigree.addedEP:
        initialisers.append('AddedEP = "{}"'.format(pedigree.addedEP))
    if pedigree.updated:
        initialisers.append('Updated = "{}"'.format(pedigree.updated))
    if pedigree.updatedEP:
        initialisers.append('UpdatedEP = "{}"'.format(pedigree.updatedEP))
    if pedigree.deprecated:
        initialisers.append('Deprecated = "{}"'.format(pedigree.deprecated))
    if pedigree.deprecatedEP:
        initialisers.append('DeprecatedEP = "{}"'.format(pedigree.deprecatedEP))
    return 'new Pedigree {{{}}}'.format(', '.join(initialisers))